ansible/provision.yaml

```diff
@@ -0,0 +1,120 @@
+
+    # If credentials are being shared between environments, generate certs except ca
+    - name: Generate cluster credentials if necessary
+      command: "{{ item }}"
+      with_items:
+        - find credentials/ ! -name 'ca*' -type f -exec rm -f {} +
+        - kube-aws render credentials
+      when: (share_cluster_credentials | lower == "y")
+
+    # If credentials are not being shared between environments, generate certs
+    - name: Generate cluster credentials if necessary
+      command: "{{ item }}"
+      with_items:
+        - rm -rf credentials/*
+        - kube-aws render credentials --generate-ca
+      when: (share_cluster_credentials | lower != "y")
+    
+    # Set the S3 bucket where the stack config is going to be stored
+    - name: Set the provisioner bucket
+      set_fact:
+        provisioner_bucket: "{{ 'k8s-provisioner-test-{{ aws_region }}' if (environment_type == 'd') else 'k8s-provisioner-prod-{{ aws_region }}' }}"
+
+    # Validate the cluster template
+    - name: Validate the cluster config
+      command: kube-aws validate --s3-uri s3://{{ provisioner_bucket }}
+      environment:
+        AWS_ACCESS_KEY_ID: "{{ aws_access_key }}"
+        AWS_SECRET_ACCESS_KEY: "{{ aws_secret_key }}"
+
+    # Provision the cluster
+    - name: Provision a cluster
+      command: kube-aws up --s3-uri s3://{{ provisioner_bucket }}
+      environment:
+        AWS_ACCESS_KEY_ID: "{{ aws_access_key }}"
+        AWS_SECRET_ACCESS_KEY: "{{ aws_secret_key }}"
+
+    # The public subnets have to tagged for the application ELB can identify the subnets that it can attach itself to
+    - name: Tag the public subnets with the clusterid
+      command: aws ec2 create-tags --resources {{ item }} --tags Key=kubernetes.io/cluster/{{ stack_name }},Value=true --region={{ region }}
+      environment:
+        AWS_ACCESS_KEY_ID: "{{ aws_access_key }}"
+        AWS_SECRET_ACCESS_KEY: "{{ aws_secret_key }}"
+      with_items:
+        - "{{ public_subnet_zone_a }}"
+        - "{{ public_subnet_zone_b }}"
+        - "{{ public_subnet_zone_c }}"
+ 
+    # Get the DNS name of the API Server ELB
+    - name: Retrieve the Cluster API server ELB
+      script: /bin/bash /get_elb.sh
+      environment:
+        AWS_REGION: "{{ region }}"
+        AWS_ACCESS_KEY_ID: "{{ aws_access_key }}"
+        AWS_SECRET_ACCESS_KEY: "{{ aws_secret_key }}"
+        CLUSTER_NAME: "{{ stack_name }}"
+      register: api_server_elb
+
+    - name: API Server ELB
+      debug:
+        msg: "API Server ELB: '{{ api_server_elb.stdout }}'"
+
+    # Create a DNS name for the API Server
+    - include: tasks/create_cname.yaml
+      vars:
+        dns_cname: "{{ stack_name }}-api"
+        hostname: "{{ api_server_elb.stdout }}"
+        
+    - include: delivery.yaml
+      when: (cluster_environment | lower == "delivery")
+
+    - include: publishing.yaml
+      when: (cluster_environment | lower == "publish")
+
+    - name: Create an S3 bucket for Splunk logs
```

Comment:
I don't remember needing this. What is it used for?

Reply:
So PeteS is working on implementing retries on Splunk forwarder. When a log fails to send to Splunk, it needs to go to its S3 bucket and the splunk forwarder comes in and picks up the logs and tries to resend them

Reply:
Nice!